security/src/main/java/no/difi/move/common/oauth/KeystoreHelper.java

```diff
@@ -0,0 +1,90 @@
+package no.difi.move.common.oauth;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.security.*;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import no.difi.asic.SignatureHelper;
+import no.difi.move.common.config.KeystoreProperties;
+
+/**
+ * Class responsible for accessing the keystore for the Integrasjonspunkt.
+ *
+ * @author Glebnn Bech
+ */
+public class KeystoreHelper {
+
+    private final KeystoreProperties keystore;
+
+    public KeystoreHelper(KeystoreProperties keystore) {
+        this.keystore = keystore;
+    }
+
+    /**
+     * Loads the private key from the keystore
+     *
+     * @return the private key
+     */
+    public PrivateKey loadPrivateKey() {
+
+        PrivateKey key = null;
+        try (InputStream i = this.keystore.getLocation().getInputStream()) {
+            KeyStore keystore = getKeystoreEntry(i);
+            key = (PrivateKey) keystore.getKey(this.keystore.getAlias(), this.keystore.getEntryPassword().toCharArray());
+            return key;
+        } catch (CertificateException | KeyStoreException | IOException | NoSuchAlgorithmException | UnrecoverableKeyException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public KeyPair getKeyPair() {
+
+        KeyPair result = null;
+        try (InputStream i = this.keystore.getLocation().getInputStream()) {
+            KeyStore keystore = getKeystoreEntry(i);
+            PrivateKey key = (PrivateKey) keystore.getKey(this.keystore.getAlias(), this.keystore.getEntryPassword().toCharArray());
+            X509Certificate c = (X509Certificate) keystore.getCertificate(this.keystore.getAlias());
+            result = new KeyPair(c.getPublicKey(), key);
+        } catch (CertificateException | KeyStoreException | IOException | NoSuchAlgorithmException | UnrecoverableKeyException e) {
+            throw new RuntimeException(e);
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS00112)